mne/io/utils.py

```diff
@@ -0,0 +1,153 @@
+                      |                        |
+                    start                    stop
+
+    We need 27 - 2 = 25 samples (per channel) to store our data, and
+    we need to read from 3 buffers (30 samples) to get all of our data.
+
+    On all reads but the first, the data we read starts at
+    the first sample of the buffer. On all reads but the last,
+    the data we read ends on the last sample of the buffer.
+
+    We call ``this_data`` the variable that stores the current buffer's data,
+    and ``data`` the variable that stores the total output.
+
+    On the first read, we need to do this::
+
+        >>> data[0:buf_len-2] = this_data[2:buf_len]  # doctest: +SKIP
+
+    On the second read, we need to do::
+
+        >>> data[1*buf_len-2:2*buf_len-2] = this_data[0:buf_len]  # doctest: +SKIP
+
+    On the final read, we need to do::
+
+        >>> data[2*buf_len-2:3*buf_len-2-3] = this_data[0:buf_len-3]  # doctest: +SKIP
+
+    This function encapsulates this logic to allow a loop over blocks, where
+    data is stored using the following limits::
+
+        >>> data[d_lims[ii, 0]:d_lims[ii, 1]] = this_data[r_lims[ii, 0]:r_lims[ii, 1]]  # doctest: +SKIP
+
+    """  # noqa
+    # this is used to deal with indexing in the middle of a sampling period
+    assert all(isinstance(x, int) for x in (start, stop, buf_len))
+    block_start_idx = (start // buf_len)
+    block_start = block_start_idx * buf_len
+    last_used_samp = stop - 1
+    block_stop = last_used_samp - last_used_samp % buf_len + buf_len
+    read_size = block_stop - block_start
+    n_blk = read_size // buf_len + (read_size % buf_len != 0)
+    start_offset = start - block_start
+    end_offset = block_stop - stop
+    d_lims = np.empty((n_blk, 2), int)
+    r_lims = np.empty((n_blk, 2), int)
+    for bi in range(n_blk):
+        # Triage start (sidx) and end (eidx) indices for
+        # data (d) and read (r)
+        if bi == 0:
+            d_sidx = 0
+            r_sidx = start_offset
+        else:
+            d_sidx = bi * buf_len - start_offset
+            r_sidx = 0
+        if bi == n_blk - 1:
+            d_eidx = stop - start
+            r_eidx = buf_len - end_offset
+        else:
+            d_eidx = (bi + 1) * buf_len - start_offset
+            r_eidx = buf_len
+        d_lims[bi] = [d_sidx, d_eidx]
+        r_lims[bi] = [r_sidx, r_eidx]
+    return block_start_idx, r_lims, d_lims
+
+
+def _read_segments_file(raw, data, idx, fi, start, stop, cals, mult,
+                        dtype='<i2', n_bytes=2):
+    """Read a chunk of raw data"""
+    n_channels = raw.info['nchan']
+
+    # data_offset and data_left count data samples (channels x time points),
+    # not bytes.
+    data_offset = n_channels * start * n_bytes
+    data_left = (stop - start) * n_channels
+
+    # Read up to 100 MB of data at a time, block_size is in data samples
+    block_size = ((100e6 // n_bytes) // n_channels) * n_channels
```

Comment:
`e` notation yields floats so I don't think this quite does what you think it will:
```
>>> type(100e6)
<type 'float'>
>>> type(100e6 // 10)
<type 'float'>
```
so better to write it out as `100000000` if you want the result to be an integer. I'm surprised there weren't new warnings thrown during tests about this (which is one reason it's important to keep our tests clear of warnings), but perhaps they just went unnoticed?